libs/vkd3d/swapchain.c

```diff
@@ -1409,6 +1412,19 @@ static bool d3d12_swapchain_has_nonzero_surface_size(struct d3d12_swapchain *swa
     return surface_caps.maxImageExtent.width != 0 && surface_caps.maxImageExtent.height != 0;
 }
 
+static void d3d12_swapchain_set_hdr_metadata(struct d3d12_swapchain* swapchain)
+{
+    const struct vkd3d_vk_device_procs *vk_procs = d3d12_swapchain_procs(swapchain);
+
+    if (!swapchain->command_queue->device->vk_info.EXT_hdr_metadata)
+        return;
+
+    if (swapchain->hdr_metadata_type == DXGI_HDR_METADATA_TYPE_NONE)
+        return;
+
+    VK_CALL(vkSetHdrMetadataEXT(swapchain->command_queue->device->vk_device, 1, &swapchain->vk_swapchain, &swapchain->hdr_metadata));
```

Comment:
This is extern-sync I believe since it modifies the swapchain state, but there's no lock in SetHDRMetadata.